pkg/services/ngalert/notifier/receiver_svc_test.go

```diff
@@ -22,101 +23,146 @@ import (
 
-		Receivers, err := sut.GetReceivers(context.Background(), rQuery(1), nil)
+		Receiver, err := sut.GetReceiver(context.Background(), singleQ(1, "slack receiver"), nil)
 		require.NoError(t, err)
-		require.Len(t, Receivers, 2)
-		require.Equal(t, "grafana-default-email", Receivers[0].Name)
-		require.Equal(t, "slack receiver", Receivers[1].Name)
+		require.Equal(t, "slack receiver", Receiver.Name)
+		require.Len(t, Receiver.GrafanaManagedReceivers, 1)
+		require.Equal(t, "UID2", Receiver.GrafanaManagedReceivers[0].UID)
 	})
 
-	t.Run("service filters receiver groups by name", func(t *testing.T) {
+	t.Run("service returns error when receiver does not exist", func(t *testing.T) {
 		sut := createReceiverServiceSut(t, secretsService)
 
-		Receivers, err := sut.GetReceivers(context.Background(), rQuery(1, "slack receiver"), nil)
-		require.NoError(t, err)
-		require.Len(t, Receivers, 1)
-		require.Equal(t, "slack receiver", Receivers[0].Name)
+		_, err := sut.GetReceiver(context.Background(), singleQ(1, "nonexistent"), nil)
+		require.ErrorIs(t, err, ErrNotFound)
 	})
 }
 
-func TestReceiverService_DecryptRedact(t *testing.T) {
+func TestReceiverService_GetReceivers(t *testing.T) {
 	sqlStore := db.InitTestDB(t)
 	secretsService := manager.SetupTestService(t, database.ProvideSecretsStore(sqlStore))
-	ac := acimpl.ProvideAccessControl(setting.NewCfg())
 
-	t.Run("service redacts receiver groups by default", func(t *testing.T) {
+	t.Run("service gets receivers from AM config", func(t *testing.T) {
 		sut := createReceiverServiceSut(t, secretsService)
-		Receivers, err := sut.GetReceivers(context.Background(), rQuery(1, "slack receiver"), nil)
-		require.NoError(t, err)
-
-		require.Len(t, Receivers, 1)
-
-		rGroup := Receivers[0]
-		require.Equal(t, "slack receiver", rGroup.Name)
-		require.Len(t, rGroup.GrafanaManagedReceivers, 1)
-
-		grafanaReceiver := rGroup.GrafanaManagedReceivers[0]
-		require.Equal(t, "UID2", grafanaReceiver.UID)
 
-		testedSettings, err := simplejson.NewJson([]byte(grafanaReceiver.Settings))
+		Receivers, err := sut.GetReceivers(context.Background(), multiQ(1), nil)
 		require.NoError(t, err)
-		require.Equal(t, definitions.RedactedValue, testedSettings.Get("url").MustString())
+		require.Len(t, Receivers, 2)
+		require.Equal(t, "grafana-default-email", Receivers[0].Name)
+		require.Equal(t, "slack receiver", Receivers[1].Name)
 	})
 
-	t.Run("service returns error when trying to decrypt with nil user", func(t *testing.T) {
+	t.Run("service filters receivers by name", func(t *testing.T) {
 		sut := createReceiverServiceSut(t, secretsService)
-		sut.ac = ac
 
-		q := rQuery(1)
-		q.Decrypt = true
-		_, err := sut.GetReceivers(context.Background(), q, nil)
-		require.ErrorIs(t, err, ErrPermissionDenied)
+		Receivers, err := sut.GetReceivers(context.Background(), multiQ(1, "slack receiver"), nil)
+		require.NoError(t, err)
+		require.Len(t, Receivers, 1)
+		require.Equal(t, "slack receiver", Receivers[0].Name)
```

Comment:
It would be great to cover exceptional cases, (db error, receiver not found), user has no permissions etc.